includes/sanitizers/class-amp-base-sanitizer.php

```diff
@@ -174,14 +174,25 @@ protected function get_body_node() {
 	 * @return float|int|string Returns a numeric dimension value, or an empty string.
 	 */
 	public function sanitize_dimension( $value, $dimension ) {
-		if ( empty( $value ) ) {
+
+		/**
+		 * Allows 0 as valid dimension.
+		 */
+		if ( null === $value ) {
 			return '';
 		}
 
 		if ( false !== filter_var( $value, FILTER_VALIDATE_INT ) ) {
 			return absint( $value );
 		}
 
+		/**
+		 * Allows floats but parses them to integers.
+		 */
+		if ( false !== filter_var( $value, FILTER_VALIDATE_FLOAT ) ) {
+			return absint( $value );
```

Comment:
Should this not be `floor( $value )`?

Reply:
@westonruter Maybe? I am looking at [the docs for floor()](http://php.net/manual/en/function.floor.php) and [round()](http://php.net/manual/en/function.round.php). Both return floats. What about `return absint( round( $value, 0 ) )`?

Reply:
`absint( -1.5 )` is `1` but `floor( -1.5 )` is `-1`. In other words, we don't want to lose the sign, do we?

Somewhat strange in the docs there for `floor()`:

> Returns the next lowest integer value (as float)

So I suppose then this is what should be done:

```php
return (int) floor( $value );
```

The `@return` phpdoc should be updated to remove `float`?

Reply:
Correction: `floor( -1.5 )` is `2`, as is `floor( -1.1 )`. Maybe `round()` is better then:

```php
return (int) round( $value, 0 );
```

Reply:
@davisshaver Hold on. I don't see how decimals are invalid AMP:

![image](https://user-images.githubusercontent.com/134745/36703829-5eecd612-1b12-11e8-80d0-ab68afd3d954.png)

Reply:
Agree about updating the `@return` statement.

I'd lean towards `absint()` to match the behavior on values that are already `int`, otherwise we may pass a negative height/width.

Reply:
True, negative numbers are illegal. But decimal-point values _are_ allowed. So then I think it should be the following with the two conditionals being combined:

```php
if ( is_numeric( $value ) ) {
    return max( 0, floatval( $value ) );
}
```

Reply:
Thanks for the discussion. I will update the PR tomorrow for another consideration